### DISCUSSION TITLE
Error in completion while in quotes (quote will often be pushed out past the end of completion and new chars added)

### DISCUSSION THREAD

**Discussion by TruePath** [community/community#15550] [Copilot Conversations]
Using RubyMine (Intellij) with github copilot has generally been amazing but whenever I try and complete code while my cursor is inside quotes inside parens it gives the wrong result.

```
console.log('NameEntry: type is Editor with value |')
                                                   ^cursor
```

The preview will look perfect.  I get

```
console.log('NameEntry: type is Editor with value ', value)
```
where , value is the (greyed out) suggested text.

But once I go to actually accept the completion it completes to 

```
console.log('NameEntry: type is Editor with value ', value)')
```

<img width="178" alt="Screen Shot 2022-04-25 at 3 23 23 AM" src="https://user-images.githubusercontent.com/121223/165041170-348ddead-8658-4aaa-80d8-963bec3c5357.png">
<img width="198" alt="Screen Shot 2022-04-25 at 3 23 28 AM" src="https://user-images.githubusercontent.com/121223/165041171-da8610e2-ba84-4797-9c3c-c76f3a9e42f1.png">

I've tried to search out other possible settings in rubymine that might be contributing but so far no luck.

**Comment by joliss** [comment]
I'm noticing this too in VS Code. It seems to happen particularly when there's closing quotes and parentheses.

For example, in a Jest test file, completing after `test("foo` (with `")` auto-inserted) causes the `")` to get repeated after the completion, leading to a stray quote mark:

<img width="210" alt="screenshot: test(&quot;foo&quot;, () => {&quot;)" src="https://user-images.githubusercontent.com/524783/191812389-ef454c31-10bf-4d28-ac79-33b322f52773.png">

This might be avoided if Copilot detected that `"` matches the beginning of the completion, and it applied some logic to delete the trailing `)`.

Interestingly, in the case of `setTimeout(`, the closing parenthesis gets deleted as it should:

<img width="347" alt="screenshot: setTimeout()   setTimeout(() => {" src="https://user-images.githubusercontent.com/524783/191813219-044fb8b6-8d6d-4c63-9f84-22936effead9.png">

which completes to

<img width="215" alt="screenshot: setTimeout(() => {" src="https://user-images.githubusercontent.com/524783/191813293-cf3016f3-09a0-43f1-84b3-e7997b558c37.png">

P.S. When suggestions repeat my code (as in `setTimeout` showing a second time in gray), it always looks confusing to me. I think I would find it less confusing if the suggestion just overdrew the auto-inserted closing parenthesis.
